extensions/amp-story/0.1/amp-story.js

```diff
@@ -19,34 +19,734 @@
+    const {width, height} = this.getViewport().getSize();
+
+    const inFullScreenThreshold =
+        width <= FULLSCREEN_THRESHOLD && height <= FULLSCREEN_THRESHOLD;
+
+    return inFullScreenThreshold && this.isAutoFullScreenEnabled_;
+  }
+
+
+  /**
+   * Handles all key presses within the story.
+   * @param {!Event} e The keydown event.
+   * @private
+   */
+  onKeyDown_(e) {
+    switch (e.keyCode) {
+      // TODO(newmuis): This will need to be flipped for RTL.
+      case KeyCodes.LEFT_ARROW:
+        this.previous_();
+        break;
+      case KeyCodes.RIGHT_ARROW:
+        this.next_();
+        break;
+    }
+  }
+
+
+  /**
+   * @param {boolean} isEnabled
+   */
+  setAutoFullScreen(isEnabled) {
+    this.isAutoFullScreenEnabled_ = isEnabled;
+  }
+
+
+  /** @private */
+  enterFullScreen_() {
+    fullscreenEnter(this.element);
+  }
+
+
+  /**
+   * @param {boolean} opt_explicitUserAction
+   * @private
+   */
+  exitFullScreen_(opt_explicitUserAction) {
+    if (opt_explicitUserAction) {
+      this.setAutoFullScreen(false);
+    }
+
+    fullscreenExit(this.element);
+  }
+
+
+  /**
+   * Invoked when the document has actually transitioned into or out of
+   * fullscreen mode.
+   * @private
+   */
+  onFullscreenChanged_() {
+    const isFullscreen = isFullscreenElement(this.element);
+    this.systemLayer_.setInFullScreen(isFullscreen);
+  }
+
+
+  /**
+   * Shows the bookend overlay.
+   * @private
+   */
+  showBookend_() {
+    if (this.isBookendActive_) {
+      return;
+    }
+
+    this.buildBookend_().then(() => {
+      this.exitFullScreen_();
+      this.systemLayer_.toggleCloseBookendButton(true);
+      this.isBookendActive_ = true;
+
+      this.vsync_.mutate(() => {
+        this.element.classList.add('i-amphtml-story-bookend-active');
+        this.bookend_.getRoot()./*OK*/scrollTop = 0;
+      });
+    });
+  }
+
+
+  /**
+   * Hides the bookend overlay.
+   * @private
+   */
+  hideBookend_() {
+    this.systemLayer_.toggleCloseBookendButton(false);
+    this.isBookendActive_ = false;
+
+    this.vsync_.mutate(() => {
+      this.element.classList.remove('i-amphtml-story-bookend-active');
+    });
+  }
+
+
+  /**
+   * Performs a system navigation if it is determined that the specified event
+   * was a click intended for navigation.
+   * @param {!Event} event 'click' event
+   * @private
+   */
+  maybePerformSystemNavigation_(event) {
+    if (!this.isNavigationalClick_(event)) {
+      // If the system doesn't need to handle this click, then we can simply
+      // return and let the event propagate as it would have otherwise.
+      return;
+    }
+
+    event.stopPropagation();
+
+    // TODO(newmuis): This will need to be flipped for RTL.
+    const nextScreenAreaMin = this.element./*OK*/offsetLeft +
+        ((1 - NEXT_SCREEN_AREA_RATIO) * this.element./*OK*/offsetWidth);
+    const nextScreenAreaMax = this.element./*OK*/offsetLeft +
+        this.element./*OK*/offsetWidth;
```

Comment:
Nit: Hoist these into local vars for readability.

```js
const offsetWidth = this.element./*OK*/offsetWidth;
```

Reply:
Done